.circleci/config.yml

```diff
@@ -186,6 +199,7 @@ executors:
     working_directory: /opt/nav2_ws
     environment:
       CMAKE_BUILD_TYPE: "Release"
```

Comment:
For a COVERAGE_ENABLED build, it'll need to be CMAKE_BUILD_TYPE=Debug to produce a useful report. With the release optimizations, the coverage data will probably be unusable.

We'll also need to set a token in the CI environment that the upload script uses. See https://codecov.io/gh/ros-planning/navigation2/settings
I believe you have access.

However, before we set that for circle CI, I think we should only upload from either Travis or Circle. Uploading twice for the same PR might be an issue. I was going to see how much of our 1000 minutes we were using before adding more to the CircleCI build. Do you know how to check the amount of time we've used?

Reply:
Nevermind. I just noticed you said the exact same thing.

Reply:
> For a COVERAGE_ENABLED build, it'll need to be CMAKE_BUILD_TYPE=Debug to produce a useful report. With the release optimizations, the coverage data will probably be unusable.

Should we just default to `Debug` builds for PRs, using a nightly cron on master to build/test both. The cron could have a forked workflow:
* testing with Debug (and thus codecove for master branch)
* testing Release (to make sure at least master is periodically Release tested)

> I was going to see how much of our 1000 minutes we were using before adding more to the CircleCI build. Do you know how to check the amount of time we've used?

I can't see this, but org admins should be able to view the stats here:  
https://circleci.com/gh/organizations/ros-planning/settings#usage

Although, open source projects (jobs for public repos I presume) shouldn't be affected though.

https://discuss.circleci.com/t/has-the-build-minutes-per-month-in-the-free-tier-been-reduced/26399/3

> However, before we set that for circle CI, I think we should only upload from either Travis or Circle.

I could comment out the codecov line in Travis to just test out Circleci. Once we merge, circleci config used for PRs would be updated, so we could then profile the added time for codecov steps.